src/mqueryfront/src/QueryMonaco.js

```diff
@@ -0,0 +1,163 @@
+import React, { Component } from "react";
+import Editor, { monaco } from "@monaco-editor/react";
+import YARA from "./yara-lang";
+
+class QueryMonaco extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            rawYara: props.rawYara,
+            editor: null,
+            readOnly: false,
+            decorations: [],
+        };
+        this.handleEditorDidMount = this.handleEditorDidMount.bind(this);
+    }
+
+    getValue() {
+        if (this.state.editor) {
+            return this.state.editor.getValue();
+        }
+    }
+
+    setValue(newValue) {
+        if (this.state.editor) {
+            this.state.editor.setValue(newValue);
+        }
+    }
+
+    setReadOnly(readOnly) {
+        if (this.state.editor) {
+            this.setState({ readOnly: readOnly });
+        }
+    }
+
+    setError(error) {
+        if (!this.state.editor || !error) {
+            return;
+        }
+
+        monaco.init().then((monaco) => {
+            this.state.decorations = this.state.editor
```

Comment:
Do not modify state directly, use `setState()`

https://reactjs.org/docs/state-and-lifecycle.html#do-not-modify-state-directly

Reply:
If you don't want to cause component re-rendering: maybe "decorations" is not a state and it should be placed in class attribute.